scripts/reporting/statsGenElastic/importOldStats.js

```diff
@@ -0,0 +1,174 @@
+*/
+
+'use strict'
+const MongoClient = require('mongodb').MongoClient;
+const Utils = require('./utils');
+const fs = require( 'fs' );
+const path = require( 'path' );
+const csv = require('csv-parser');
+
+var errors = [];
+
+try {
+	if ( Utils.clean(process.env.ELASTIC_CLOUD_AUTH === undefined ) ) { errors.push(new Error('ELASTIC_CLOUD_AUTH is not set')); } 
+	if ( Utils.clean(process.env.ELASTIC_CLOUD_ID === undefined ) ) { errors.push(new Error('ELASTIC_CLOUD_ID is not set')); }
+	if ( errors.length > 0 ) { throw errors }
+} catch (e) {
+    for (let i=0; i<e.length; i++) {
+		console.log(e[i].message);
+	  }
+	process.exit(-1);
+}
+
+try {
+	process.env.ELASTIC_CLOUD_AUTH.split(":")
+} catch (error) {
+	console.log("Error: Failed to split ELASTIC_CLOUD_AUTH, check format " + process.env.ELASTIC_CLOUD_AUTH)
+	process.exit(-1);
+}
+
+process.on("unhandledRejection", (error) => {
+	console.error(error); // This prints error with stack included (as for normal errors)
+	throw error; // Following best practices re-throw error and let the process exit with error code
+  })
+
+const { Client } = require('@elastic/elasticsearch')
+const ELASTIC_CLOUD_AUTH = process.env.ELASTIC_CLOUD_AUTH.split(":")
+const ElasticClient = new Client({
+	cloud: {
+	  id: process.env.ELASTIC_CLOUD_ID
+	},
+	auth: {
+	  username: ELASTIC_CLOUD_AUTH[0],
+	  password: ELASTIC_CLOUD_AUTH[1]
+	},
+	reload_connections: true,
+	maxRetries: 5,
+	request_timeout: 60
+  });
+  
+ElasticClient.cluster.health({},function(err,resp,status) {  
+	console.log("[ELASTIC] -- Client Health --",resp);
+});
+
+if(process.argv.length < 5) {
+	console.error('Not enough arguments: \n\tUsage: node importOldStats.js <mongo host:port> <username> <password> <import-csv-folder>');
+	process.exit(-1);
+}
+
+try {
+	if ( Utils.clean(process.argv[5] === undefined ) ) { errors.push(new Error('import-csv-folder is not set')); } 
+	if ( errors.length > 0 ) { throw errors }
+} catch (e) {
+    for (let i=0; i<e.length; i++) {
+		console.log(e[i].message);
+	  }
+	process.exit(-1);
+}
+
+const url = `mongodb://${process.argv[3]}:${process.argv[4]}@${process.argv[2]}/admin`;
+const client = new MongoClient(url, { useUnifiedTopology: true });
+const importFolder = process.argv[5]
+let opt  = {delimiter: ',', quote: '\'', escape: '\'', relax: true, skip_empty_lines: true};
+
+start();
+
+async function start() {
+	console.log(`Trying to connect to ${url}...`);
+	try {
+		const db = await client.connect();
+		console.log('Connected successfully!');
+
+		(async ()=>{
+			try {
+				// Get the files as an array
+				const files = await fs.promises.readdir( importFolder );
+				const seen = [] // array to catch previously seen id's
+
+				for( const file of files ) {
+					const fullPath = path.join( importFolder, file );
+					const stat = await fs.promises.stat( fullPath );
+					
+					if( stat.isFile() && file.includes(".csv") && !file.includes("com.dropbox.attrs")  ){
```

Comment:
more reliable to check for endsWith (assuming .csv is still going to be an extension)
```suggestion
					if( stat.isFile() && file.endsWith(".csv") && !file.includes("com.dropbox.attrs")  ){
```

Reply:
done